### PR TITLE
HW_6 "OOP"

### PR DESCRIPTION
**Pull request by rudneff13**

### CODE REVIEW

File: oop/homework.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+            self.saturation_level = [self._increase_saturation_level(v)
+                                     for k, v in product_dict.items() if product == k][0]
+        return self.saturation_level
+
+    def _reduce_saturation_level(self, value):
+        self.saturation_level -= value
+        if self.saturation_level < 0:
+            self.saturation_level = 0
+            return self.saturation_level
+        else:
```
Comment (dimigor) [inline oop/homework.py]: Do you really need this else block? You can simply one-time use return after if block
Also, you don't need return, I did not see where you give a variable result of execution this method

**Review by dimigor — CHANGES_REQUESTED** [review]

File: oop/homework.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+            self.average_speed = 90
+        if 5 < self.age <= 15:
+            self.average_speed = 75
+        if self.age > 15:
+            self.average_speed = 40
+        return self.average_speed
+
+    def __str__(self):
+        return f"""
+    Current cheetah is {self.age} years old.
```
Comment (dimigor) [inline oop/homework.py]: Make formatting prettier

File: oop/homework.py [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+       * Implement method window_square which return result of simple square formula of rectangle
+
+    """
+
+    def __init__(self, width, height):
+        self.width = width
+        self.height = height
+
+    def window_square(self):
+        return self.width * self.height
```
Comment (dimigor) [inline oop/homework.py:214]: What happens when width or height will be 0?

File: oop/homework.py [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+
+    def __init__(self):
+        self.__walls = []
+        self.__windows = []
+        self.__roof = None
+        self.__door = None
+
+    def create_wall(self, width, height):
+        if len(self.__walls) < 4:
+            if (width and height) == 0:
```
Comment (dimigor) [inline oop/homework.py]: `and` or `or`?

File: oop/homework.py (hunk 1 above)
Comment (rudneff13) [reply oop/homework.py]: block else removed as much as possible

File: oop/homework.py (hunk 4 above)
Comment (rudneff13) [reply oop/homework.py]: you are right. or.

File: oop/homework.py (hunk 3 above)
Comment (rudneff13) [reply oop/homework.py:214]: window_square = 0
:)

File: oop/homework.py (hunk 2 above)
Comment (rudneff13) [reply oop/homework.py]: done

**Review by dimigor — APPROVED** [review]

**Pull request merged**